tests/unit/commands/samconfig/test_samconfig.py

```diff
@@ -0,0 +1,483 @@
+        Path(self.scratch_dir, "envvar.json").write_text("{}")
+
+        os.chdir(self.scratch_dir)
+
+    def tearDown(self):
+        os.chdir(self._old_cwd)
+        shutil.rmtree(self.scratch_dir)
+        self.scratch_dir = None
+
+    @patch("samcli.commands.init.do_cli")
+    def test_init(self, do_cli_mock):
+        config_values = {
+            "no_interactive": True,
+            "location": "github.com",
+            "runtime": "nodejs10.x",
+            "dependency_manager": "maven",
+            "output_dir": "myoutput",
+            "name": "myname",
+            "app_template": "apptemplate",
+            "no_input": True,
+            "extra_context": '{"key": "value", "key2": "value2"}',
+        }
+
+        with samconfig_parameters(["init"], self.scratch_dir, **config_values) as config_path:
+            from samcli.commands.init import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(
+                ANY,
+                True,
+                "github.com",
+                "nodejs10.x",
+                "maven",
+                "myoutput",
+                "myname",
+                "apptemplate",
+                True,
+                '{"key": "value", "key2": "value2"}',
+            )
+
+    @patch("samcli.commands.validate.validate.do_cli")
+    def test_validate(self, do_cli_mock):
+        config_values = {"template_file": "mytemplate.yaml"}
+
+        with samconfig_parameters(["validate"], self.scratch_dir, **config_values) as config_path:
+
+            from samcli.commands.validate.validate import cli
+
+            LOG.debug(Path(config_path).read_text())
+            runner = CliRunner()
+            result = runner.invoke(cli, [])
+
+            LOG.info(result.output)
+            LOG.info(result.exception)
+            if result.exception:
+                LOG.exception("Command failed", exc_info=result.exc_info)
+            self.assertIsNone(result.exception)
+
+            do_cli_mock.assert_called_with(ANY, str(Path(os.getcwd(), "mytemplate.yaml")))
+
+    @patch("samcli.commands.build.command.do_cli")
+    def test_build(self, do_cli_mock):
+        config_values = {
+            "function_identifier": "foo",
+            "template_file": "mytemplate.yaml",
+            "base_dir": "basedir",
+            "build_dir": "builddir",
+            "use_container": True,
+            "manifest": "requirements.txt",
+            "docker_network": "mynetwork",
+            "skip_pull_image": True,
+            "parameter_overrides": "ParameterKey=Key,ParameterValue=Value ParameterKey=Key2,ParameterValue=Value2",
```

Comment:
can we add parameter overrides that are "a=b c=d" as well, that way we exercise the second regex.

Reply:
Yes, some of the others have this